tilings/strategies/assumption_splitting.py

```diff
@@ -0,0 +1,136 @@
+from typing import Iterator, List, Optional, Tuple
+
+from sympy import Eq, Function
+
+from comb_spec_searcher import CombinatorialObject, Constructor, Strategy
+from comb_spec_searcher.strategies.constructor import (
+    RelianceProfile,
+    SubGens,
+    SubRecs,
+    SubSamplers,
+)
+from tilings import GriddedPerm, Tiling
+from tilings.algorithms import Factor
+from tilings.assumptions import TrackingAssumption
+
+
+class Split(Constructor):
+    def is_equivalence(self) -> bool:
+        return False
+
+    def get_equation(self, lhs_func: Function, rhs_funcs: Tuple[Function, ...]) -> Eq:
+        raise NotImplementedError
+
+    def reliance_profile(self, n: int, **parameters: int) -> RelianceProfile:
+        raise NotImplementedError
+
+    def get_recurrence(self, subrecs: SubRecs, n: int, **parameters: int) -> int:
+        raise NotImplementedError
+
+    def get_sub_objects(
+        self, subgens: SubGens, n: int, **parameters: int
+    ) -> Iterator[Tuple[CombinatorialObject, ...]]:
+        raise NotImplementedError
+
+    def random_sample_sub_objects(
+        self,
+        parent_count: int,
+        subsamplers: SubSamplers,
+        subrecs: SubRecs,
+        n: int,
+        **parameters: int
+    ):
+        raise NotImplementedError
+
+    @staticmethod
+    def get_eq_symbol() -> str:
+        return "↣"
+
+
+class SplittingStrategy(Strategy[Tiling, GriddedPerm]):
+    @staticmethod
+    def decomposition_function(tiling: Tiling) -> Optional[Tuple[Tiling]]:
+        if not tiling.assumptions:
+            return None
+        components = Factor(tiling.underlying_tiling()).get_components()
+        if len(components) == 1:
+            return None
+        new_assumptions: List[TrackingAssumption] = []
+        for ass in tiling.assumptions:
+            assert isinstance(
+                ass, TrackingAssumption
+            ), "not implemented splitting assumption for given type"
+            split_gps: List[List[GriddedPerm]] = [[] for _ in range(len(components))]
+            for gp in ass.gps:
+                for idx, component in enumerate(components):
+                    if all(cell in component for cell in gp.pos):
+                        split_gps[idx].append(gp)
+                        # only add to one list
+                        break
+                else:
+                    # gridded perms can't be partitioned
+                    new_assumptions.extend([ass])
```

Comment:
The purpose of this function is to look at each assumptions, and partition them into "sub"  assumptions, according to the factors: 

i.e. if the assumption is {1: (0, 0), 1: (1, 1)} then this will be turned into {1: (0, 0)} and {1: (1, 1)}. This will allow us to factor, as the original assumption will forbid the separating (0, 0) and (1, 1). 

> should consider the factor of the gp

Not in this function, but this should be a separate strategy, as the way to count it will be different.